cdptools/bin/clone_file_store.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import logging
+from pathlib import Path
+
+from google.api_core.page_iterator import Page
+
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+import cdptools.file_stores.gcs_file_store as fs
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s"
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class Args(argparse.Namespace):
+
+    def __init__(self):
+        self.__parse()
+
+    def __parse(self):
+        p = argparse.ArgumentParser(
+            prog="clone_file_store",
+            description="Clone source file store to target file store."
+        )
+
+        p.add_argument("--target_bucket", type=str, help="Target bucket name.", required=True)
+        p.add_argument("--target_credentials", type=Path, help="Credentials of target file store.", required=True)
+        p.add_argument("--debug", action="store_true", dest="debug", help="Show traceback if the script were to fail.")
+        p.add_argument('--source_bucket', help="Source bucket name.", required=True)
+        p.add_argument('--source_credentials', type=Path, help="Credentials of source file store.")
+
+        p.parse_args(namespace=self)
+
+
+def clone_page(page: Page, source_fs: fs.GCSFileStore, target_fs: fs.GCSFileStore):
+    for blob in page:
+        # download file takes in a filename and returns a path
+        downloaded_path = source_fs.download_file(blob.name, overwrite=True)
+
+        # upload file takes in a path
+        target_fs.upload_file(downloaded_path)
+
+
+def main():
+    args = Args()
+
+    source_fs = fs.GCSFileStore(args.source_bucket, args.source_credentials)
+    target_fs = fs.GCSFileStore(args.target_bucket, args.target_credentials)
+
+    # We call client list_blobs method as use of the bucket's list_blobs method is deprecated
+    source_client = source_fs._client
+    source_bucket = source_fs._bucket
```

Comment:
These lines are really specific to CloudFirestoreDatabase. Anyway these can be generalized more to allow for AppDirs and future dbs?

Reply:
How generalized do we want to make it? Do you mean making the script itself be able to be run wit any file store without any code modifications? Or do you mean just swapping out whatever file store you plan on using in the code itself?